src/gui/controllers/TaskController.java

```diff
@@ -18,5 +22,25 @@
 	public void initialize(URL location, ResourceBundle resources) {
 
 	}
+	
+	@FXML public void accept(ActionEvent event) {
+		Object userData = textFieldTitle.getScene().getWindow().getUserData();
+		if (isValidData() && userData != null && userData instanceof TasksHandler) {
+			try {
+				((TasksHandler) userData).addTask(new Task(textFieldTitle.getText(),textAreaBody.getText()));
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+		Utils.closeWindow(event);
+	}
+
+	@FXML public void cancel(ActionEvent event) {
+		Utils.closeWindow(event);
+	}
+	
+	private boolean isValidData() {
+		return textFieldTitle.getText().length() > 0;
```

Comment:
Why are you not validating if the textAreaBody has content? Are we going to let users save tasks with just a title?

Reply:
You're right, I forgot to validate the textarea.
Actually, we have to validate the date too, if the task is a reminder.

Reply:
I still have doubts about how to do that.